tmt/checks/coredump.py

```diff
@@ -0,0 +1,614 @@
+
+
+COREDUMP_LAST_DUMP_FILENAME = "coredump-latest"
+
+# Can be set in /etc/coredumpct.conf.d/
+# See `man coredump.conf`
+COREDUMP_CONFIG = """[Coredump]
+Storage=external
+Compress=yes
+"""
+
+
+@container
+class CoredumpCheck(Check):
+    """Configuration for the coredump check."""
+
+    # Patterns to ignore in crash reports
+    ignore_patterns: list[Pattern[str]] = field(
+        default_factory=list,
+        help="""
+             Optional list of regular expressions to ignore in crash reports.
+             If a crash report matches any of these patterns, it will be ignored
+             and not cause a failure. Any other crashes will still cause the test
+             to fail. If no patterns are specified, any crash will cause a failure.
+             """,
+        metavar="PATTERN",
+        normalize=tmt.utils.normalize_pattern_list,
+        exporter=lambda patterns: [pattern.pattern for pattern in patterns],
+        serialize=lambda patterns: [pattern.pattern for pattern in patterns],
+        unserialize=lambda serialized: [re.compile(pattern) for pattern in serialized],
+    )
+
+    # Internal flag to track if we can run coredumpctl on the host
+    is_available: bool = True
+    is_availability_reason: Optional[str] = None
+
+    # Path to the file storing information about coredumps before test execution
+    # Default is current directory, will be properly set in _save_existing_coredumps
+    coredump_last_dumps_filepath: Path = field(default_factory=Path, internal=True)
+
+    def to_spec(self) -> _RawCheck:
+        """Convert to raw specification."""
+        spec = super().to_spec()
+
+        spec["ignore-patterns"] = [  # type: ignore[reportGeneralTypeIssues,typeddict-unknown-key,unused-ignore]
+            pattern.pattern for pattern in self.ignore_patterns
+        ]
+
+        return spec
+
+    def to_minimal_spec(self) -> _RawCheck:
+        """Convert to minimal raw specification."""
+        return self.to_spec()
+
+    def _configure_coredump(self, guest: "Guest", logger: tmt.log.Logger) -> None:
+        """
+        Try configure coredump storage.
+
+        Non-privileged users might not have permission to change the config,
+        while being able to use coredumpctl
+        """
+        try:
+            guest.execute(
+                ShellScript("mkdir -p /etc/systemd/coredump.conf.d")
+                & ShellScript(
+                    f"echo '{COREDUMP_CONFIG}' > /etc/systemd/coredump.conf.d/50-tmt.conf"
+                )
+            )
+            return
+        except tmt.utils.RunError:
+            logger.debug("Unable to configure coredump directly, trying with sudo")
+
+        # If failed and not root, try with sudo
+        if guest.facts.is_superuser is False:
+            try:
+                guest.execute(
+                    ShellScript("sudo mkdir -p /etc/systemd/coredump.conf.d")
+                    & ShellScript(
+                        f"echo '{COREDUMP_CONFIG}' | sudo tee"
```

Comment:
Missing space. Without it, the script would look like this: `...sudo tee/etc/...`
```suggestion
                        f"echo '{COREDUMP_CONFIG}' | sudo tee "
```

Reply:
fixed, thanks!